### PR TITLE
[CI:DOCS] Add more documentation on conmon

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman.1.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ The Network File System (NFS) and other distributed file systems (for example: L
 For more information, please refer to the [Podman Troubleshooting Page](https://github.com/containers/podman/blob/master/troubleshooting.md).
 
 ## SEE ALSO
-`containers-mounts.conf(5)`, `containers-registries.conf(5)`, `containers-storage.conf(5)`, `buildah(1)`, `containers.conf(5)`, `oci-hooks(5)`, `containers-policy.json(5)`, `crun(8)`, `runc(8)`, `subuid(5)`, `subgid(5)`, `slirp4netns(1)`
+`containers-mounts.conf(5)`, `containers-registries.conf(5)`, `containers-storage.conf(5)`, `buildah(1)`, `containers.conf(5)`, `oci-hooks(5)`, `containers-policy.json(5)`, `crun(8)`, `runc(8)`, `subuid(5)`, `subgid(5)`, `slirp4netns(1)`, , **conmon**(8).
```
Comment (zhangguanzhang) [inline docs/source/markdown/podman.1.md:346]:
```suggestion
`containers-mounts.conf(5)`, `containers-registries.conf(5)`, `containers-storage.conf(5)`, `buildah(1)`, `containers.conf(5)`, `oci-hooks(5)`, `containers-policy.json(5)`, `crun(8)`, `runc(8)`, `subuid(5)`, `subgid(5)`, `slirp4netns(1)`,`conmon(8)`.
```

Comment (rhatdan) [reply docs/source/markdown/podman.1.md]: Went a slightly different direction.

File: docs/source/markdown/podman-create.1.md [hunk 2]
```diff
@@ -1412,6 +1412,18 @@ $ podman start --attach ctr
...
+## CONMON
+
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program who’s job is to watch the primary process of the container, and if the
+container dies, save the exit code.  It also holds open the tty of the
+container, so that it can be attached to later. This is what allows Podman to
+run in detached mode (backgrounded), so Podman can exit but conmon continues to
+run.  Each container has their own instance of conmon. When Podman interacts
+with the container (stop, exec, restart ...), it is actually communicating with
```
Comment (mheon) [inline docs/source/markdown/podman-create.1.md:1424]: Not true - we're communicating with the OCI runtime, which doesn't interact with Conmon at all. Conmon is really just sitting, waiting for an exit, while we deal directly with container PID1

File: docs/source/markdown/podman-create.1.md [hunk 3]
```diff
@@ -1412,6 +1412,18 @@ $ podman start --attach ctr
 b
 ```
 
+## CONMON
+
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program who’s job is to watch the primary process of the container, and if the
```
Comment (eriksjolund) [inline docs/source/markdown/podman-create.1.md:1419]:
```suggestion
program whose job is to watch the primary process of the container, and if the
```

File: docs/source/markdown/podman-run.1.md [hunk 4]
```diff
@@ -1763,6 +1763,18 @@ $ podman run --env ENV*****=b alpine printenv ENV*****
 b
 ```
 
+## CONMON
+
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program who’s job is to watch the primary process of the container, and if the
```
Comment (eriksjolund) [inline docs/source/markdown/podman-run.1.md:1770]:
```suggestion
program whose job is to watch the primary process of the container, and if the
```

File: docs/source/markdown/podman-create.1.md (hunk 2 above)
Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md:1424]: Give me better description then.

Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md]: Check it out now.

File: docs/source/markdown/podman-create.1.md [hunk 5]
```diff
@@ -1412,6 +1412,19 @@ $ podman start --attach ctr
...
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program whose job is to watch the primary process of the container, and if the
+container dies, save the exit code.  It also holds open the tty of the
+container, so that it can be attached to later. This is what allows Podman to
+run in detached mode (backgrounded), so Podman can exit but conmon continues to
+run.  Each container has their own instance of conmon. Conmon waits for the
+container to exit, gathers and saves the exit code, and then launches a Podman
+process to complete the container cleanup, by shutting down the network and
+storage.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:1426]: Maybe?
```suggestion
storage.  For more information on common, please reference the common(8) man page.
```
Or something to that effect.

File: docs/source/markdown/podman-run.1.md [hunk 6]
```diff
@@ -1763,6 +1763,19 @@ $ podman run --env ENV*****=b alpine printenv ENV*****
...
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program whose job is to watch the primary process of the container, and if the
+container dies, save the exit code.  It also holds open the tty of the
+container, so that it can be attached to later. This is what allows Podman to
+run in detached mode (backgrounded), so Podman can exit but conmon continues to
+run.  Each container has their own instance of conmon. Conmon waits for the
+container to exit, gathers and saves the exit code, and then launches a Podman
+process to complete the container cleanup, by shutting down the network and
+storage.
+
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1778]: If you take the maybe suggestion above, please add here too.

**Comment by TomSweeneyRedHat** [comment]
One suggestion for consideration.  If @mheon is hip with the new wording, then LGTM

File: docs/source/markdown/podman-create.1.md (hunk 2 above)
Comment (mheon) [reply docs/source/markdown/podman-create.1.md]: Works for me

File: docs/source/markdown/podman-create.1.md [hunk 7]
```diff
@@ -1412,6 +1412,20 @@ $ podman start --attach ctr
...
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program whose job is to watch the primary process of the container, and if the
+container dies, save the exit code.  It also holds open the tty of the
+container, so that it can be attached to later. This is what allows Podman to
+run in detached mode (backgrounded), so Podman can exit but conmon continues to
+run.  Each container has their own instance of conmon. Conmon waits for the
+container to exit, gathers and saves the exit code, and then launches a Podman
+process to complete the container cleanup, by shutting down the network and
+storage.   For more information on common, please reference the common(8) man
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:1426]:
```suggestion
storage.   For more information on conmon, please reference the common(8) man
```

File: docs/source/markdown/podman-run.1.md [hunk 8]
```diff
@@ -1763,6 +1763,20 @@ $ podman run --env ENV*****=b alpine printenv ENV*****
...
+When Podman starts a container it actually executes the conmon program, which
+then executes the OCI Runtime.  Conmon is the container monitor.  It is a small
+program whose job is to watch the primary process of the container, and if the
+container dies, save the exit code.  It also holds open the tty of the
+container, so that it can be attached to later. This is what allows Podman to
+run in detached mode (backgrounded), so Podman can exit but conmon continues to
+run.  Each container has their own instance of conmon. Conmon waits for the
+container to exit, gathers and saves the exit code, and then launches a Podman
+process to complete the container cleanup, by shutting down the network and
+storage.   For more information on common, please reference the common(8) man
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1777]:
```suggestion
storage.   For more information on conmon, please reference the conmon(8) man
```

File: docs/source/markdown/podman-create.1.md (hunk 7 above)
Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md]: I blame you for that, since I just cut and pased your previous message, at least I believe I did.

**Comment by TomSweeneyRedHat** [comment]
LGTM
BTB, I'm happy to see I'm not the only one who types "common" instead of "common"

**Pull request merged**